sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/exception/CosmosExceptionUtils.java

```diff
@@ -37,9 +38,40 @@ public static <T> Mono<T> exceptionHandler(String message, Throwable throwable,
+                case HttpConstants.StatusCodes.FORBIDDEN -> throw new CosmosForbiddenException(message, unwrappedThrowable);
+                case HttpConstants.StatusCodes.GONE-> {
+                    switch(((CosmosException) unwrappedThrowable).getSubStatusCode()) {
+                        case HttpConstants.SubStatusCodes.NAME_CACHE_IS_STALE -> throw new CosmosInvalidPartitionException(message, unwrappedThrowable);
+                        case HttpConstants.SubStatusCodes.COMPLETING_PARTITION_MIGRATION -> throw new CosmosPartitionIsMigratingException(message, unwrappedThrowable);
+                        case HttpConstants.SubStatusCodes.PARTITION_KEY_RANGE_GONE -> throw new CosmosPartitionKeyRangeGoneException(message, unwrappedThrowable);
+                        case HttpConstants.SubStatusCodes.COMPLETING_SPLIT_OR_MERGE -> throw new CosmosPartitionKeyRangeIsSplittingException(message, unwrappedThrowable);
+                        default -> throw new CosmosGoneException(message, unwrappedThrowable);
+                    }
+                }
+                case HttpConstants.StatusCodes.INTERNAL_SERVER_ERROR -> throw new CosmosInternalServerErrorException(message, unwrappedThrowable);
+                case HttpConstants.StatusCodes.METHOD_NOT_ALLOWED -> throw new CosmosMethodNotAllowedException(message, unwrappedThrowable);
+                case HttpConstants.StatusCodes.NOTFOUND -> throw new CosmosNotFoundException(message, unwrappedThrowable);
+                case HttpConstants.StatusCodes.REQUEST_TIMEOUT -> {
+                    if (((CosmosException) unwrappedThrowable).getSubStatusCode() == HttpConstants.SubStatusCodes.CLIENT_OPERATION_TIMEOUT) {
+                        throw new CosmosOperationCancelledException(message, unwrappedThrowable);
+                    }
+                    throw new CosmosRequestTimeoutException(message, unwrappedThrowable);
+                }
+                case HttpConstants.StatusCodes.PRECONDITION_FAILED -> throw new CosmosPreconditionFailedException(message, unwrappedThrowable);
+                case HttpConstants.StatusCodes.REQUEST_ENTITY_TOO_LARGE -> throw new CosmosRequestEntityTooLargeException(message, unwrappedThrowable);
+                case HttpConstants.StatusCodes.TOO_MANY_REQUESTS -> throw new CosmosRequestRateTooLargeException(message, unwrappedThrowable);
+                case HttpConstants.StatusCodes.RETRY_WITH -> throw new CosmosRetryWithException(message, unwrappedThrowable);
+                case HttpConstants.StatusCodes.SERVICE_UNAVAILABLE -> throw new CosmosServiceUnavailableException(message, unwrappedThrowable);
+                case HttpConstants.StatusCodes.UNAUTHORIZED -> throw new CosmosUnauthorizedException(message, unwrappedThrowable);
+                default -> throw new CosmosAccessException(message, unwrappedThrowable);
+            }
+
         }
         throw new CosmosAccessException(message, unwrappedThrowable);
```

Comment:
If the inner exception is not a CosmosException - should we really wrap this in a CosmosAccessException? I don't think we would ever expect this?

Reply:
@kushagraThapar thoughts on this? This is what we had always done so this was left as the default.

Reply:
@FabianMeiswinkel I am curious, why not throw the `unwrappedThrowable` as part of the `CosmosAccessException`? Even if the inner exception is not a `CosmosException`, we can still throw just the `CosmosAccessException` - and that way, it won't be a breaking change either, since we were doing this before.

Reply:
This is in an if-statement where we ensured that the unwrappedThrowable is of type COmsosException. Technically either is the same - but passing the cosmsoException as inner IMO makes it easier to read/understand
Any concerns with that?

Reply:
No concerns with that, I didn't see the if statement, I think we can close this discussion. All good!